src/amplify-migration/yamlUtils.ts

```diff
@@ -43,17 +43,19 @@ export async function changeContentInYamlFile(
     filePath = newPermalink;
   }
 
-  if (setOfAllDocumentsPath.has(filePath.toLowerCase())) {
-    // YAML does not seem to have a way to update the value of a key in place
-    fileContent = fileContent.replace(oriFilePath, filePath.toLowerCase());
-  } else {
+  // YAML does not seem to have a way to update the value of a key in place
+  // We just mutate all to lowercase to not care about encoding. Then we report if image is not found
+  // rather than programmatically fixing something that we are not 100% sure of.
+  fileContent = fileContent.replace(oriFilePath, filePath.toLowerCase());
```

Comment:
fix 1.2

Reply:
does `setOfAllDocumentsPath` come with lowercased items by default? otherwise when we the `.has` method, it might be that it was the original version but not the lower cased one

Reply:
> does setOfAllDocumentsPath come with lowercased items by default?

good qn, and yes, we convert everything to lowercase!

Reply:
we can consider using a utility [type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html#lowercasestringtype) here so it's clear to devs + this invariant is maintained.

if this file isn't going to be actively maintained, then it's ok otherwise, it seems important enough (and maybe low effort enough?) to do

Reply:
ok have done it [here](https://github.com/isomerpages/isomer-conversion-scripts/pull/70/commits/5b354298a5737236c59b2da35f2cd20ced86be81)